### DISCUSSION TITLE
optimized spend - limit channel  budget of optimal spend

### DISCUSSION THREAD

**Discussion by ohad-monday** [google/lightweight_mmm#127] [Ideas]
in the demo example, let's say i want channel_0 to stay in the same level (200) of the spend for some reason.
optimization with default lower/upper limits:
<img width="1003" alt="image" src="https://user-images.githubusercontent.com/61185809/209547747-d3d68aeb-1d87-4366-8304-7dc7fcdf09a4.png">

I tried using the lower/upper bounds and limit channel_0 with (lower/upper limits are 0 for channel_0):
import jax.numpy as jnp
upper = jnp.asarray([0,0.9,0.9])
lower = jnp.asarray([0,0.9,0.9])

solution, kpi_without_optim, previous_media_allocation = optimize_media.find_optimal_budgets(
    n_time_periods=n_time_periods,
    media_mix_model=mmm,
    extra_features=extra_features_scaler.transform(extra_features_test)[:n_time_periods],
    budget=budget,
    prices=prices,
    bounds_lower_pct = lower,
    bounds_upper_pct = upper,
    media_scaler=media_scaler,
    target_scaler=target_scaler,
    seed=SEED)

optimization with custom lower/upper limits, show lower decrease in  but still a decrease (150) but i want it to stay the same:
<img width="999" alt="image" src="https://user-images.githubusercontent.com/61185809/209548114-51b8fc74-543b-4feb-9c02-d4da3f77065e.png">

am i missing something here?

**Comment by michevan** [comment]
Hi! I just tried replicating this using the mock data in the national-level example colab, and got a different result. Are you still seeing this issue?


![example](https://user-images.githubusercontent.com/32819754/210821276-7bda3081-04de-4202-9091-ec76f5f45d0e.png)

**Reply by ohad-monday** [reply]
yes, still same issue.
i'm using a docker image with lightweight_mmm-0.1.5 - i will try maybe to update to a newer version

**Reply by michevan** [reply]
sounds good, let us know if that helps! We're up to v0.1.7 now as of today

**Reply by ohad-monday** [reply]
it works! with an updated version

**Reply by michevan** [reply]
great to hear, thank you for the update!
